compiler/src/iree/compiler/Codegen/Dialect/VectorExt/IR/VectorExtOps.cpp

```diff
@@ -37,6 +38,407 @@ OpFoldResult ToSIMTOp::fold(FoldAdaptor) {
   return {};
 }
 
+//
+// TransferGatherOp
+//
+
+Speculation::Speculatability TransferGatherOp::getSpeculatability() {
+  if (isa<RankedTensorType>(getSource().getType())) {
+    return Speculation::Speculatable;
+  }
+  return Speculation::NotSpeculatable;
+}
+
+void TransferGatherOp::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  if (llvm::isa<MemRefType>(getSource().getType())) {
+    effects.emplace_back(MemoryEffects::Read::get(), &getSourceMutable(),
+                         SideEffects::DefaultResource::get());
+  }
+}
+
+const char *kNonIndexedSymbol = "None";
+
+ParseResult
+parseIndexVecs(OpAsmParser &parser,
+               SmallVectorImpl<OpAsmParser::UnresolvedOperand> &indexVecs,
+               SmallVectorImpl<Type> &indexVecTypes, ArrayAttr &indexed) {
+  if (parser.parseLSquare())
+    return failure();
+
+  SMLoc loc;
+  SmallVector<bool> indexedArr;
+  while (parser.parseOptionalRSquare()) {
+    // Check if this dimension is not indexed
+    if (!parser.parseOptionalKeyword(kNonIndexedSymbol)) {
+      indexedArr.push_back(false);
+      (void)parser.parseOptionalComma();
+      continue;
+    }
+
+    OpAsmParser::UnresolvedOperand indexVec;
+    Type indexVecType;
+    if (parser.getCurrentLocation(&loc) || parser.parseOperand(indexVec) ||
+        parser.parseColon() || parser.parseType(indexVecType)) {
+      return parser.emitError(loc, "Expected `none` or `operand : type`");
+    }
+
+    indexVecs.push_back(indexVec);
+    indexVecTypes.push_back(indexVecType);
+    indexedArr.push_back(true);
+
+    (void)parser.parseOptionalComma();
+  }
+
+  // OpBuilder is only used as a helper to build an BoolArrayAttr.
+  OpBuilder b(parser.getContext());
```

Comment:
nit: just a builder should be good enough here.
```suggestion
  Builder b(parser.getContext());
```